ckan/templates/package/resource_read.html

```diff
@@ -43,6 +44,24 @@
                     <i class="fa fa-arrow-circle-o-down"></i> {{ _('Download') }}
                   {% endif %}
                 </a>
+                {%if res.url_type == 'upload' and res.datastore_active %}
```

Comment:
Don't check for `res.url_type == 'upload'` you can have stuff in the DataStore that hasn't been uploaded